container_workflow_tool/distgit.py

```diff
@@ -464,13 +477,14 @@ def push_changes(self, tmp, images):
             component = image["component"]
             try:
                 repo = Repo(component)
-                self.logger.info("Pushing: " + component)
-                # If a commit message is provided do a commit first
-                if self.commit_msg and repo.is_dirty():
-                    # commit_msg is set so it is always returned
-                    commit = self.get_commit_msg(None, image)
-                    repo.git.commit("-am", commit)
-                repo.git.push()
+                if self._get_unpushed_commits(repo):
+                    self.logger.info("Pushing: " + component)
+                    # If a commit message is provided do a commit first
+                    if self.commit_msg and repo.is_dirty():
+                        # commit_msg is set so it is always returned
+                        commit = self.get_commit_msg(None, image)
+                        repo.git.commit("-am", commit)
+                    repo.git.push()
```

Comment:
It would be nice to have an `else:` clause here for debugging output. Something like "Push skipped for $image as there is nothing to push."